src/main/java/com/amazon/opendistroforelasticsearch/ad/task/ADBatchTaskRunner.java

```diff
@@ -177,6 +196,193 @@ public ADBatchTaskRunner(
 
         this.pieceIntervalSeconds = BATCH_TASK_PIECE_INTERVAL_SECONDS.get(settings);
         clusterService.getClusterSettings().addSettingsUpdateConsumer(BATCH_TASK_PIECE_INTERVAL_SECONDS, it -> pieceIntervalSeconds = it);
+
+        this.maxTopEntitiesPerHcDetector = MAX_TOP_ENTITIES_FOR_HISTORICAL_ANALYSIS.get(settings);
+        clusterService
+            .getClusterSettings()
+            .addSettingsUpdateConsumer(MAX_TOP_ENTITIES_FOR_HISTORICAL_ANALYSIS, it -> maxTopEntitiesPerHcDetector = it);
+
+        this.maxRunningEntitiesPerDetector = MAX_RUNNING_ENTITIES_PER_DETECTOR_FOR_HISTORICAL_ANALYSIS.get(settings);
+        clusterService
+            .getClusterSettings()
+            .addSettingsUpdateConsumer(MAX_RUNNING_ENTITIES_PER_DETECTOR_FOR_HISTORICAL_ANALYSIS, it -> maxRunningEntitiesPerDetector = it);
+    }
+
+    /**
+     * Run AD task on worker node.
+     * 1. For HC detector, will get top entities first. If top entities already initialized,
```

Comment:
Does this happen on worker node or coordinating node? I thought get top entities happen on the coordinating node.

Reply:
yes, correct, fixed java doc in new PR